srml/elections-phragmen/src/lib.rs

```diff
@@ -0,0 +1,1259 @@
+	fn is_voter(who: &T::AccountId) -> bool {
+		<StakeOf<T>>::exists(who)
+	}
+
+	/// The locked stake of a voter.
+	fn locked_stake_of(who: &T::AccountId) -> BalanceOf<T> {
+		Self::stake_of(who)
+	}
+
+	/// Check there's nothing to do this block.
+	///
+	/// Runs phragmen election and cleans all the previous candidate state. The voter state is NOT
+	/// cleaned and voters must themselves submit a transaction to retract.
+	fn end_block(block_number: T::BlockNumber) -> dispatch::Result {
+		if (block_number % Self::term_duration()).is_zero() {
+			Self::do_phragmen();
+		}
+		Ok(())
+	}
+
+	/// Run the phragmen election with all required side processes and state updates.
+	///
+	/// Calls the appropriate `ChangeMembers` function variant internally.
+	fn do_phragmen() {
+		let desired_seats = Self::desired_members() as usize;
+		let desired_runner_ups = Self::desired_runner_ups() as usize;
+		let num_to_elect = desired_runner_ups + desired_seats;
+
+		// current members are always a candidate for the next round as well.
+		// this is guaranteed to not create any duplicates.
+		let mut candidates = Self::candidates();
+		candidates.append(&mut Self::members());
+		let voters_and_votes = <VotesOf<T>>::enumerate()
+			.map(|(v, i)| (v, i.0))
+			.collect::<Vec<(T::AccountId, Vec<T::AccountId>)>>();
+		let maybe_new_members = phragmen::elect::<_, _, _, T::CurrencyToVote>(
+			num_to_elect,
+			false,
+			0,
+			candidates.clone(),
+			voters_and_votes,
+			Self::locked_stake_of
+		);
+
+		let mut to_release_bond: Vec<T::AccountId> = Vec::with_capacity(desired_seats);
+		let old_members = <Members<T>>::take();
+		if let Some((mut new_members_with_approval, _)) = maybe_new_members {
+			// sort based on stake to to split winner and runner ups.
+			new_members_with_approval.sort_by_key(|x| x.1.clone());
+
+			// filter out those who had literally no votes at all.
+			let new_members_filtered = new_members_with_approval
+				.into_iter()
+				.filter_map(|(m, a)| if !a.is_zero() { Some(m) } else { None } )
+				.collect::<Vec<T::AccountId>>();
+
+			let mut new_members = new_members_filtered.iter()
+				.rev()
+				.take(desired_seats)
+				.map(|m| m.clone())
+				.collect::<Vec<T::AccountId>>();
+			new_members.sort();
+			<Members<T>>::put(new_members.clone());
+
+			let mut allowed_runner_ups = new_members_filtered.len() - new_members.len();
+			// defensive only. We should never have more leftover than desired runner ups.
+			allowed_runner_ups = allowed_runner_ups.min(desired_runner_ups);
+			let runner_ups = new_members_filtered.into_iter()
+				.take(allowed_runner_ups)
+				.collect::<Vec<T::AccountId>>();
+			// should still be sorted by stake. Low to high.
+			<RunnerUps<T>>::put(runner_ups.clone());
+
+			// report. We don't know/compute the diff here, therefore we call `set_member`.
+			let (incoming, outgoing) = T::ChangeMembers::compute_members_diff(&new_members, &old_members);
+			T::ChangeMembers::change_members_sorted(
```

Comment:
why unrolling the API? just use `set_members_sorted`.

Reply:
Because I use the diff which is computed within `set_member_sorted` somewhere a bit further down and --call me selfish for changing `ChangeMember` according to my uses case-- I think it is better than duplicating the code that computes the diff of incoming/outgoing set.